src/Servers/Kestrel/Core/src/Internal/Http/Http1Connection.cs

```diff
@@ -275,9 +275,29 @@ private void OnOriginFormTarget(HttpMethod method, HttpVersion version, Span<byt
             try
             {
                 // Read raw target before mutating memory.
-                RawTarget = target.GetAsciiStringNonNullCharacters();
-                QueryString = query.GetAsciiStringNonNullCharacters();
-                Path = PathNormalizer.DecodePath(path, pathEncoded, RawTarget, query.Length);
```

Comment:
You've got to admit this one is artificial. What scenarios besides benchmarks hit the same URL back-to-back at any scale sufficient for this optimization to be relevant?

Reply:
Lots of users hitting F5 on an overloaded webserver

Reply:
Lots of users hitting F5 on an overloaded webserver; get latest news on a site not doing real-time, to check stock on a sold out item hoping it comes back, trying to book tickets for a popular gig that's just gone on sale, etc

Reply:
It's fine as long as it doesn't regress performance, that's why the optimizing compare is important.

Reply:
It has increased complexity and thus reduced maintainability.

Even when you mash F5 it's going to make several requests for different resources, and spread them across different connections. You'll rarely get the same request back-to-back.

Reply:
>  You'll rarely get the same request back-to-back.

Will do if your resources are in CDN; they'll be on a different domain, different "servers"

Reply:
>  You'll rarely get the same request back-to-back.

Will do if your resources are in CDN; they'll be on a different domain, different "servers"

Example: for the url https://www.ageofascent.com/2019/02/04/asp-net-core-saturating-10gbe-at-7-million-requests-per-second/ even though there are 87 requests that make up the page, there is always only a single request to `www.ageofascent.com` which contains the page body, everything else is from other domains.

Reply:
It’s fine. This doesn’t cross the line for me and doesn’t  make the code that much worse. IMO The part of this change that is concerning is the byte comparsion.

Reply:
> What scenarios besides benchmarks hit the same URL back-to-back at any scale sufficient for this optimization to be relevant?

Long polling uses same path and multiple requests, range requests use the same url and changes the Range header; REST apis based on POST body values, e.g. GraphQL uses a single url and then changes body data; Webhooks will typically POST to a fixed URL, etc.

Although; you raise an important gap, the most common request of a website is for the homepage, that being OriginForm `/`; so I have added an optimization to use an interned string of ForwardSlash even on the first request; where there has been no prior request.